### PR TITLE
ci: add script to check modules on registry.coder.com

### PR DESCRIPTION
**Pull request by johnstcn**
I made a simple and dumb script to check the modules on registry.coder.com.
Gonna see how it goes for the moment before hooking it up to any sad trombones.

### CODE REVIEW

File: check.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+done
+echo "Checking modules: ${modules[*]}"
+for module in "${modules[@]}"; do
+    # Trim leading/trailing whitespace from module name
+    module=$(echo "${module}" | xargs)
+    url="${REGISTRY_BASE_URL}/modules/${module}"
+    printf "=== Check module %s at %s\n" "${module}" "${url}"
+    status_code=$(curl --output /dev/null --head --silent --fail --location "${url}" --retry 3 --write-out "%{http_code}")
+    # shellcheck disable=SC2181
+    if (( status_code != 200 )); then
```
Comment (dannykopping) [inline check.sh:24]: I wonder if the server returns `3XX` cache responses?

**Review by dannykopping — APPROVED** [review]
LGTM!

File: check.sh (hunk 1 above)
Comment (johnstcn) [reply check.sh:24]: I haven't noticed any of those after running it but I suppose the proof is in the eating.

**Review by matifali — APPROVED** [review]
A few suggestions but LGTM.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Check modules on registry.coder.com
+
+on:
+  schedule:
+    - cron: "23 * * * *" # Runs at 23 minutes past the hour.
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply .github/workflows/check.yaml:5]: Yep! I'd pick a prime number though. Lucky 13?

**Pull request merged**